### PR TITLE
Asyncify verbose option

### PR DESCRIPTION
**Pull request by kripken**
This logs out the decisions made about instrumenting functions, which
can help figure out why a function is instrumented, or to get a list of
what might need to be.

As the test shows, it can print things like this:
```
[asyncify] import is an import that can change the state
[asyncify] calls-import can change the state due to import
[asyncify] calls-calls-import can change the state due to calls-import
[asyncify] calls-calls-calls-import can change the state due to calls-calls-import
```
(the test has `calls-calls-calls-import => calls-calls-import => calls-import -> import`).

Not sure how useful this might be, but it was easy to write up. Thoughts?

cc @curiousdannii @aardappel

### CODE REVIEW

File: src/passes/Asyncify.cpp [hunk 1]
```diff
@@ -663,6 +691,11 @@ class ModuleAnalyzer {
           if (matched) {
             info.addedFromList = true;
           }
+          if (verbose) {
+            std::cout << "[asyncify] " << func->name
+                      << "'s state is set based on the only-list to " << matched
+                      << '\n';
+          }
```
Comment (aheejin) [inline src/passes/Asyncify.cpp:698]: Is looks possible for some functions to be logged as added as can-change-state functions but if the only-list is present, they are eventually ignored silently, i.e., without any other log messages that they are ignored, right? Would it be better to check the only list when logging for other causes..?

Comment (kripken) [reply src/passes/Asyncify.cpp:698]: I think this log message here would be shown, though, so it's not completely silent? This message would say they are set to `false`, which overrides the previous messages.

Comment (aheejin) [reply src/passes/Asyncify.cpp:698]: Ah you're right. Sorry I thought this is only printed when the value is true.

**Comment by aardappel** [comment]
Yes, the "due to" will be super useful in tracking down why a function that doesn't appear necessary is included, thanks!

**Pull request merged**